### ISSUE TITLE
Implement scf workchain failure handling

### ISSUE THREAD

**Issue by Tseplyaev** [JuDFTteam/aiida-fleur#54]
Scf workchian should probably use RestartBaseWorkChain to handle some failures.

Examples:

1. Retry: try to converge restarting from this calculation, i.e. if scf is ended and did not lead to convergence, resubmit it.
2. ggf handle bad convergence behavior, delete broyd (do not copy), i.e. try to change mixing scheme, factors...

**Comment by Tseplyaev** [comment]
RestartBaseWorkChain was shown to work with other WorkChains, not only CalcJobs. Thus it can be implemented.

**Comment by broeder-j** [comment]
I would not like to have an additional workchain, but rather to we should adapt the scf workchain in such a way that it has does this.

**Comment by Tseplyaev** [comment]
I agree. In case of SCF workchain it would be better to implement the logic inside the workchain. This will help us to reduce the number of generated DB nodes.
However, I am still finding the RestartWorkChain functionality pretty convenient in implementation. It also shows the logic better: if I pass some input parameter into a RestartWorkChain I expect them to be adjusted if something goes wrong.
In contrast, if I give something inside SCF workchain, I want these parameters to not change. In this case we would have a clear logical separation between failing and successful input setups. 
In my opinion, we should decide what is more important for us: clearer logic and easier development (both are actually arguable) or the size of the DB.
